src/pages/FlagCommentPage.js

```diff
@@ -161,7 +162,18 @@ function FlagCommentPage(props) {
         >
             {({safeAreaPaddingBottomStyle}) => (
                 <FullPageNotFoundView shouldShow={!ReportUtils.shouldShowFlagComment(getActionToFlag(), props.report)}>
-                    <HeaderWithBackButton title={props.translate('reportActionContextMenu.flagAsOffensive')} />
+                    <HeaderWithBackButton
+                        title={props.translate('reportActionContextMenu.flagAsOffensive')}
+                        onBackButtonPress={() => {
+                            const topMostReportID = Navigation.getTopmostReportId();
```

Comment:
@suneox I can see this is exactly same as `ReportDetailsPage`. Why not move this to a util? so that we can use it again?

Reply:
Can we modify the condition so that `topMostReportID` is fetched only if `shouldAwareTopMostReport` is set true?